anchor/tests/create_escrow_test.spec.ts

```diff
@@ -0,0 +1,213 @@
+    name.write(testName)
+
+    // Escrow Description (200 bytes)
+    const testDescription = 'TEST_ESCROW description'
+    const description = Buffer.alloc(200)
+    description.write(testDescription)
+
+    const [escrowKey, _bump] = anchor.web3.PublicKey.findProgramAddressSync(
+      [Buffer.from('escrow'), escrowId],
+      program.programId
+    )
+
+    await program.methods.createEscrow(
+      Array.from(escrowId),
+      name,
+      description
+    )
+    .accounts({
+      escrow: escrowKey,
+      signer: initialiser.publicKey,
+      systemProgram: anchor.web3.SystemProgram.programId,
+    })
+    .signers([initialiser])
+    .rpc()
+
+    const escrowAccount = await program.account.escrow.fetch(escrowKey)
+    expect(Buffer.from(escrowAccount.id).toString('hex')).toBe(uuid)
+    expect(escrowAccount.initialiser.toBase58()).toBe(initialiser.publicKey.toBase58())
+
+    // Convert the bytes to strings and remove trailing null characters
+    const storedName = Buffer.from(escrowAccount.name).toString().replace(/\0/g, '')
+    const storedDescription = Buffer.from(escrowAccount.description).toString().replace(/\0/g, '')
+
+    expect(storedName).toBe(testName)
+    expect(storedDescription).toBe(testDescription)
+    expect(escrowAccount.totalAmount.toNumber()).toBe(0)
+    expect(escrowAccount.isPublicDeposit).toBe(true)
+    expect(escrowAccount.depositorsCount).toBe(0)
+    expect(escrowAccount.recipientsCount).toBe(0)
+    expect(escrowAccount.status).toEqual({ draft: {} })
+    expect(escrowAccount.modules).toEqual([])
+  })
+
+  it('Same UUID should fail', async () => {
+    // This test is to check if the same UUID can be used to create two different escrows
+    // Should FAIL
+
+    const uuid = uuidv4().replace(/-/g, '')
+    const escrowId = Uint8Array.from(Buffer.from(uuid, 'hex'))
+
+    const [escrowKey, _bump] = anchor.web3.PublicKey.findProgramAddressSync(
+      [Buffer.from('escrow'), escrowId],
+      program.programId
+    )
+
+    // Escrow Name (100 bytes)
+    const testName = 'TEST_ESCROW name'
+    const name = Buffer.from(testName, 'utf8')
+
+    // Escrow Description (200 bytes)
+    const testDescription = 'TEST_ESCROW description'
+    const description = Buffer.from(testDescription, 'utf8')
+
+    // First Escrow Creation
+    await program.methods.createEscrow(
+      Array.from(escrowId),
+      name,
+      description
+    )
+    .accounts({
+      escrow: escrowKey,
+      signer: initialiser.publicKey,
+      systemProgram: anchor.web3.SystemProgram.programId,
+    })
+    .signers([initialiser])
+    .rpc()
+
+    // Second Escrow Creation
+    await program.methods.createEscrow(
+      Array.from(escrowId),
+      name,
+      description
+    )
+    .accounts({
+      escrow: escrowKey,
+      signer: initialiser.publicKey,
+      systemProgram: anchor.web3.SystemProgram.programId,
+    })
+    .signers([initialiser])
+    .rpc()
+    .catch((error) => {
+      expect(error).toBeDefined()
```

Comment:
we should declare a variable and check this outside the catch otherwise if there is no error we won't see the test failing.